s390x: CI broken by "bump go-patricia to 2.2.4"?

Looks like PR #25283 (commit  3d714b5ed58cfdfd5872ddd3654d171b09bb02d3) introduced a problem for s390x. See Jenkins run: https://jenkins.dockerproject.org/job/Docker%20Master%20(z)/132

```
FAIL: docker_cli_build_test.go:4318: TestBuildEntrypointInheritance.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_build_test.go:507: TestBuildOnBuildCmdEntrypointJSON.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_api_containers_test.go:1225: TestContainerApiGetContainersJSONEmpty.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_api_containers_test.go:88: TestContainerApiPsOmitFields.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_health_test.go:53: TestHealth.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_import_test.go:16: TestImportDisplay.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_import_test.go:45: TestImportFile.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_import_test.go:92: TestImportFileWithMessage.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_import_test.go:67: TestImportGzipped.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_info_test.go:125: TestInfoDisplaysPausedContainers.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_info_test.go:114: TestInfoDisplaysRunningContainers.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_info_test.go:140: TestInfoDisplaysStoppedContainers.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_links_test.go:32: TestLinksPingLinkedContainers.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_port_test.go:275: TestPortExposeHostBinding.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_port_test.go:253: TestPortHostBinding.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_port_test.go:14: TestPortList.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_ps_test.go:19: TestPsListContainersBase.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_ps_test.go:263: TestPsListContainersFilterAncestorImage.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_ps_test.go:403: TestPsListContainersFilterExited.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_rm_test.go:52: TestRmContainerOrphaning.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_run_test.go:4352: TestRunAttachFailedNoLeak.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_run_test.go:2988: TestRunContainerWithRmFlagCannotStartContainer.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_run_test.go:2971: TestRunContainerWithRmFlagExitCodeNotEqualToZero.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_run_test.go:4319: TestRunNamedVolumeNotRemoved.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_run_test.go:4336: TestRunNamedVolumesFromNotRemoved.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_run_test.go:2958: TestRunVolumesFromRestartAfterRemoved.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_port_test.go:180: TestUnpublishedPortsInPsOutput.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_update_unix_test.go:132: TestUpdateKernelMemoryUninitialized.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_volume_test.go:240: TestVolumeCliCreateWithOpts.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_volume_test.go:66: TestVolumeCliLs.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_volume_test.go:98: TestVolumeCliLsFilterDangling.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_volume_test.go:173: TestVolumeCliRm.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_api_volumes_test.go:66: TestVolumesApiInspect.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_api_volumes_test.go:13: TestVolumesApiList.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_api_volumes_test.go:42: TestVolumesApiRemove.pN52_github_com_docker_docker_integration_cli.DockerSuite
FAIL: docker_cli_daemon_test.go:797: TestDaemonBridgeFixedCidr2.pN58_github_com_docker_docker_integration_cli.DockerDaemonSuite
FAIL: docker_cli_daemon_test.go:2048: TestDaemonRestartWithNames.pN58_github_com_docker_docker_integration_cli.DockerDaemonSuite
FAIL: docker_cli_daemon_test.go:81: TestDaemonRestartWithVolumesRefs.pN58_github_com_docker_docker_integration_cli.DockerDaemonSuite
FAIL: docker_cli_daemon_test.go:2257: TestRunLinksChanged.pN58_github_com_docker_docker_integration_cli.DockerDaemonSuite
FAIL: docker_api_swarm_test.go:601: TestApiSwarmLeaveRemovesContainer.pN57_github_com_docker_docker_integration_cli.DockerSwarmSuite
FAIL: docker_api_swarm_test.go:907: TestApiSwarmRestartCluster.pN57_github_com_docker_docker_integration_cli.DockerSwarmSuite
FAIL: docker_api_swarm_test.go:207: TestApiSwarmServicesCreate.pN57_github_com_docker_docker_integration_cli.DockerSwarmSuite
FAIL: docker_cli_external_volume_driver_unix_test.go:411: TestExternalVolumeDriverBindExternalVolume.pN66_github_com_docker_docker_integration_cli.DockerExternalVolumeSuite
FAIL: docker_cli_external_volume_driver_unix_test.go:440: TestExternalVolumeDriverGet.pN66_github_com_docker_docker_integration_cli.DockerExternalVolumeSuite
FAIL: docker_cli_external_volume_driver_unix_test.go:426: TestExternalVolumeDriverList.pN66_github_com_docker_docker_integration_cli.DockerExternalVolumeSuite
FAIL: docker_cli_external_volume_driver_unix_test.go:294: TestExternalVolumeDriverVolumesFrom.pN66_github_com_docker_docker_integration_cli.DockerExternalVolumeSuite
```

Investigating...

@unclejack : I now tried the "go-patricia" testsuite on my laptop (amd64) with gccgo 6.1. It works with version v2.1.0 and fails with version v2.2.4. So it seems to be an architecture independent gccgo related problem.

Here the steps to reproduce:

1) Start gcc:6.1 image and get go-patricia code
```
$ docker run -ti --rm gcc:6.1 bash
root@a0f013b0c10a:~# mkdir -p /root/go/src
root@a0f013b0c10a:~# cd /root/go/src
root@a0f013b0c10a:~/go/src# export GOPATH=/root/go
root@a0f013b0c10a:~/go/src# go get github.com/satori/go.uuid
root@a0f013b0c10a:~/go/src# git clone https://github.com/tchap/go-patricia.git github.com/tchap/go-patricia
```

2) Try with version 2.1.0 (works):
```
root@a0f013b0c10a:~/go/src# cd github.com/tchap/go-patricia/patricia && git checkout v2.1.0 && cd -
Previous HEAD position was 0d76659... Fix Delete memory leak and other bugs found
HEAD is now at d87fea6... Merge pull request #14 from ishidawataru/fix_walk
root@a0f013b0c10a:~/go/src# go test github.com/tchap/go-patricia/patricia
ok      github.com/tchap/go-patricia/patricia   4.646s
```

2) Try with version 2.2.4 (fails):
```
root@a0f013b0c10a:~/go/src# cd github.com/tchap/go-patricia/patricia && git checkout v2.2.4 && cd -
HEAD is now at 0d76659... Fix Delete memory leak and other bugs found
root@a0f013b0c10a:~# go test github.com/tchap/go-patricia/patricia

--- FAIL: TestTrie_DeleteLeakageDense (0.00s)
panic: runtime error: index out of range [recovered]
        panic: runtime error: index out of range


        :0

        :0

github_com_tchap_go_patricia_patricia.remove.pN53_github_com_tchap_go_patricia_patricia.sparseChildList
        /root/go/src/github.com/tchap/go-patricia/patricia/children.go:73
github_com_tchap_go_patricia_patricia.Delete.pN42_github_com_tchap_go_patricia_patricia.Trie
        /root/go/src/github.com/tchap/go-patricia/patricia/patricia.go:302
github_com_tchap_go_patricia_patricia.TestTrie_DeleteLeakageDense
        /root/go/src/github.com/tchap/go-patricia/patricia/patricia_dense_test.go:196

        :0

goroutine 16 [chan receive]:

        :0


created by main
        /usr/src/gcc/libgo/runtime/go-main.c:54

        :0

        :0

goroutine 17 [syscall]:
        goroutine in C code; stack unavailable

goroutine 18 [finalizer wait]:

goroutine 19 [GC sweep wait]:
        :0

        :0
FAIL    github.com/tchap/go-patricia/patricia   0.044s
```

--- a/hack/vendor.sh
+++ b/hack/vendor.sh
@@ -53,7 +53,7 @@ clone git github.com/gorilla/mux e444e69cbd
 clone git github.com/kr/pty 5cf931ef8f
 clone git github.com/mattn/go-shellwords v1.0.0
 clone git github.com/mattn/go-sqlite3 v1.1.0
-clone git github.com/tchap/go-patricia v2.2.4
+clone git github.com/tchap/go-patricia v2.2.5
 clone git github.com/vdemeester/shakers 24d7f1d6a71aa5d9cbe7390e4afb66b7eef9e1b3
 # forked golang.org/x/net package includes a patch for lazy loading trace templates
 clone git golang.org/x/net 2beffdc2e92c8a3027590f898fe88f69af48a3f8 https://github.com/tonistiigi/net.git

